cfme/tests/infrastructure/test_instance_analysis.py

```diff
@@ -0,0 +1,299 @@
+# -*- coding: utf-8 -*-
+# These tests don't work at the moment, due to the security_groups multi select not working
+# in selenium (the group is selected then immediately reset)
+import fauxfactory
+import pytest
+import cfme.fixtures.pytest_selenium as sel
+
+from cfme.common.vm import VM
+from cfme.common.provider import cleanup_vm
+from cfme.infrastructure import host
+from cfme.infrastructure.pxe import get_template_from_config
+from cfme.provisioning import do_vm_provisioning
+from cfme.web_ui import InfoBlock, Table, paginator
+from datetime import datetime
+from fixtures.pytest_store import store
+from utils import testgen, ssh
+from utils.conf import cfme_data
+from utils.version import current_version
+from utils.update import update
+from utils.wait import wait_for
+
+pytestmark = [pytest.mark.meta(server_roles="+automate +smartproxy +smartstate")]
+
+
+def pytest_generate_tests(metafunc):
+    # Filter out providers without templates defined
+    argnames, argvalues, idlist = testgen.infra_providers(metafunc, 'provisioning')
+    argnames = argnames + ['cloud_init_template']
+
+    new_argvalues = []
+    new_idlist = []
+    for i, argvalue_tuple in enumerate(argvalues):
+        args = dict(zip(argnames, argvalue_tuple))
+        if not args['provisioning']:
+            # Don't know what type of instance to provision, move on
+            continue
+
+        if not {'ci-template', 'ci-username', 'ci-pass', 'ci-image'}.issubset(
+                args['provisioning'].viewkeys()):
+            # Need all for template provisioning
+            continue
+
+        cloud_init_template = args['provisioning']['ci-template']
+        if cloud_init_template not in cfme_data.get('customization_templates', {}).keys():
+            continue
+
+        argvalues[i].append(get_template_from_config(cloud_init_template))
+
+        new_idlist.append(idlist[i])
+        new_argvalues.append(argvalues[i])
+
+    testgen.parametrize(metafunc, argnames, new_argvalues, ids=new_idlist, scope="module")
+
+
+@pytest.fixture(scope="module")
+def setup_provider(request, setup_provider_modscope, provider, provisioning):
+    if provider.type == 'virtualcenter':
+        store.current_appliance.install_vddk(reboot=True)
+        store.current_appliance.wait_for_web_ui()
+        try:
+            sel.refresh()
+        except AttributeError:
+            # In case no browser is started
+            pass
+        from cfme import login
+        login.login_admin()
+        set_host_credentials(request, provisioning, provider)
```

Comment:
If the previous code is to make sure we are logged in in order to "set_host_credentials", then that function should have a force_navigate at the start. We shouldn't have to log in here. a much simpler way of doing it, if indeed we need to, is to sel.force_navigate('<<whatever the host_credentials page is>>')